components/motor/gpio/motor_encoder_test.go

```diff
@@ -4,755 +4,279 @@ import (
-			pos = int64(posFl)
-			test.That(tb, pos, test.ShouldEqual, 1)
-			test.That(tb, err, test.ShouldBeNil)
-		})
-
-		test.That(t, encA.Tick(context.Background(), false, nowNanosTest()), test.ShouldBeNil) // 11 -> 10
-		testutils.WaitForAssertion(t, func(tb testing.TB) {
-			tb.Helper()
-			pos := enc.RawPosition()
-			test.That(tb, pos, test.ShouldEqual, 1)
-			posFl, _, err := enc.Position(context.Background(), encoder.PositionTypeUnspecified, nil)
-			pos = int64(posFl)
-			test.That(tb, pos, test.ShouldEqual, 0)
-			test.That(tb, err, test.ShouldBeNil)
-		})
-
-		test.That(t, encB.Tick(context.Background(), false, nowNanosTest()), test.ShouldBeNil) // 10 -> 00
-		testutils.WaitForAssertion(t, func(tb testing.TB) {
-			tb.Helper()
-			pos := enc.RawPosition()
-			test.That(tb, pos, test.ShouldEqual, 0)
-			posFl, _, err := enc.Position(context.Background(), encoder.PositionTypeUnspecified, nil)
-			pos = int64(posFl)
-			test.That(tb, pos, test.ShouldEqual, 0)
+			on, powerPct, err := m.IsPowered(context.Background(), nil)
+			test.That(tb, on, test.ShouldBeTrue)
+			test.That(tb, powerPct, test.ShouldBeLessThan, 0)
 			test.That(tb, err, test.ShouldBeNil)
 		})
 	})
 
-	t.Run("motor encoder test GoFor (forward)", func(t *testing.T) {
-		th := setup(t)
-		defer th.Teardown()
-
-		encA := th.EncoderA
-		encB := th.EncoderB
-		realMotor := th.RealMotor
-		motor := th.Motor
-
-		err := motor.goForInternal(context.Background(), 100, 1, 1)
-		test.That(t, err, test.ShouldBeNil)
+	t.Run("encoded motor test goForMath", func(t *testing.T) {
 		testutils.WaitForAssertion(t, func(tb testing.TB) {
 			tb.Helper()
-			test.That(tb, realMotor.Direction(), test.ShouldEqual, 1)
+			test.That(tb, m.ResetZeroPosition(context.Background(), 0, nil), test.ShouldBeNil)
 		})
 
-		err = motor.goForInternal(context.Background(), -100, 1, 1)
-		test.That(t, err, test.ShouldBeNil)
-		test.That(t, realMotor.Direction(), test.ShouldEqual, 1)
+		expectedGoalPos, expectedGoalRPM, expectedDirection := 4.0, 10.0, 1.0
+		goalPos, goalRPM, direction := m.goForMath(context.Background(), 10, 4)
 
 		testutils.WaitForAssertion(t, func(tb testing.TB) {
 			tb.Helper()
-			test.That(tb, realMotor.PowerPct(), test.ShouldBeGreaterThan, 0.0)
+			test.That(tb, goalPos, test.ShouldEqual, expectedGoalPos)
+			test.That(tb, goalRPM, test.ShouldEqual, expectedGoalRPM)
+			test.That(tb, direction, test.ShouldEqual, expectedDirection)
 		})
 
-		for x := 0; x < 100; x++ {
-			test.That(t, encB.Tick(context.Background(), true, nowNanosTest()), test.ShouldBeNil)
-			test.That(t, encA.Tick(context.Background(), true, nowNanosTest()), test.ShouldBeNil)
-			test.That(t, encB.Tick(context.Background(), false, nowNanosTest()), test.ShouldBeNil)
-			test.That(t, encA.Tick(context.Background(), false, nowNanosTest()), test.ShouldBeNil)
-		}
+		expectedGoalPos, expectedGoalRPM, expectedDirection = -4.0, -10.0, -1.0
+		goalPos, goalRPM, direction = m.goForMath(context.Background(), 10, -4)
```

Comment:
[note to self] ah so this ties into the other comment. go for math changes the rpm to match the direction, and shifts the position to be used with the encoder